### PR TITLE
boards: Add JLink support for samr21-xpro board

### PR DESCRIPTION
**Pull request by kaifeichen**
boards: Add JLink support for samr21-xpro board

- add a PROGRAMMER flag (values: edbg, jlink)

@Hyungsin @immesys

### CODE REVIEW

File: boards/samr21-xpro/Makefile.include [hunk 1]
```diff
@@ -2,7 +2,29 @@
...
+
+  export FLASHER = $(RIOTBOARD)/$(BOARD)/dist/flash.sh
+  export FFLAGS  = $(BINDIR) $(HEXFILE)
+
+  export OFLAGS    = -O binary --gap-fill 0xff
+  export HEXFILE = $(ELFFILE:.elf=.bin)
+  export DEBUGGER_FLAGS = $(BINDIR) $(ELFFILE)
+  export RESET_FLAGS = $(BINDIR)
+
+  export TERMPROG = JLinkRTTClient
```
Comment (immesys) [inline boards/samr21-xpro/Makefile.include]: Using rtt_stdio is separate from flashing with JTAG, you may want to flash with JTAG but still use the serial for stdio

File: boards/samr21-xpro/dist/reset.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+# This script resets a CC2538SF53 target using JLink called
```
Comment (immesys) [inline boards/samr21-xpro/dist/reset.sh]: I think at some point there was talk of having a central standard version of these scripts, if so, you should use those and if not you should update this comment to not say CC2538

File: boards/samr21-xpro/Makefile.include (hunk 1 above)
Comment (immesys) [reply boards/samr21-xpro/Makefile.include]: Maybe put a guard here and only specify this if rtt_stdio is in USE_MODULEs

**Comment by smlng** [comment]
I'm with @immesys, if Jlink is used for multiple boards, respective scripts should somehow be centralised in `/dist/tools`, and board specific jlink config params for instance for samr21-xpro into the existing `sam0.inc.mk` (if possible?).

**Comment by immesys** [comment]
@gebart, the biggest advantage is that jlinkgdbserver supports SEGGER RTT but at the time that I evaluated it (~2 years ago) the JLink tooling was also much faster and more responsive than the openOCD tooling. I know a lot can change in two years though, so that may not be true anymore.

**Comment by kbumsik** [comment]
I personally use [SEGGER Ozone](https://www.segger.com/products/development-tools/ozone-j-link-debugger/) and this debugger is the most fantastic debugger I’ve seen.

**Pull request closed without merging**